Added ruleId deprecation notice (#1078)
codatio/codat-docs: blog/240319-deprecation-ruleId.md

---
title: "2024-07-10: Deprecation of ruleId in rules webhooks"
date: "2024-03-19"
tags: ["Deprecation"]
authors: dcoplowe
---

> On **July 10, 2024**, we will deprecate the `ruleId` property in all our existing rule payloads following the release of our [new webhooks service](/updates/240306-new-webhook-service-released).

Reviewer: It is unclear whether this only impacts the user once the migrate to the new service

Reviewer 2: any value in mentioning webhooks in the sentence? like, "in all our existing webhook payloads" or something

<!--truncate-->

> The `ruleId` property will be deprecated in the following webhook rule schemas:

Reviewer: `RuleId` is capital R in our docs

Reviewer 2: yeah, I believe that's how it is spelled in the original responses etc

- [Push operation status has changed](https://docs.codat.io/using-the-api/webhooks/legacy/core-rules-types#push-operation-status-has-changed)
- [Push operation has timed out](https://docs.codat.io/using-the-api/webhooks/legacy/core-rules-types#push-operation-has-timed-out)
- [Account categories updated](https://docs.codat.io/using-the-api/webhooks/legacy/core-rules-types#account-categories-updated)
- [Sync Connection Deleted](https://docs.codat.io/using-the-api/webhooks/legacy/core-rules-types#sync-connection-deleted)
- [Expenses sync completed](https://docs.codat.io/using-the-api/webhooks/legacy/core-rules-types#expenses-sync-completed)
- [Expenses sync failed](https://docs.codat.io/using-the-api/webhooks/legacy/core-rules-types#expenses-sync-failed)
- [Client rate limit reached](https://docs.codat.io/using-the-api/webhooks/legacy/core-rules-types#client-rate-limit-reached)
- [Client rate limit reset](https://docs.codat.io/using-the-api/webhooks/legacy/core-rules-types#client-rate-limit-reset)

Following the release of the new webhooks service, the `ruleId` is no longer relevant and should not be used.   

### Action required

> Review and update your application logic to remove any dependencies on the `ruleId`. This will help prevent any disruptions to your integration with Codat.

Reviewer: ```suggestion
Review and update your application logic to remove any dependencies on the `RuleId`. This will help prevent any disruptions to your integration with Codat.

You should use `RuleType` to identify what event a given webhook corresponds with.
```

### Expected impact if no action is taken

> If no action is taken by the deprecation date, your application will continue to recieve events however the `ruleId` will not be populated.

Reviewer 2: ```suggestion
If no action is taken, after **July 10, 2024**, your application will continue to receive events, but the `RuleId` will not be populated.
```

> Any application logic using the `ruleId` property will function following your unexpected `ruleId` logic.

Reviewer 2: what does "following your unexpected ruleid logic" mean?

Reviewer 2: should this say

Any application logic that uses the `ruleId` property may function in an unexpected manner.

> Depending on your implementation this would likely cause your application to stop processing Codat's webhooks. 

Reviewer 2: ```suggestion
Depending on your implementation, this will likely cause your application to stop processing Codat's webhooks. 
```
